### PR TITLE
add ProcessorStateManagerTest

### PR DESCRIPTION
**Pull request by ymatsuda**
* added ProcessorStateManagerTest
* bug fix: correctly unassign a changelog partition (instead of unsubscribing)

@guozhangwang

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java [hunk 1]
```diff
@@ -163,8 +164,8 @@ public void register(StateStore store, RestoreFunc restoreFunc) {
         long newOffset = restoreConsumer.position(storePartition);
         restoredOffsets.put(storePartition, newOffset);
 
-        // un-subscribe the change log partition
-        restoreConsumer.subscribe(Collections.<String>emptyList());
+        // un-assign the change log partition
+        restoreConsumer.assign(Collections.<TopicPartition>emptyList());
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java]: There is a new API added along with regex subscription patch:

unsubscribe()

which unsubscribed all current subscriptions, could you merge from apache trunk and use this function instead?

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+            private final Deserializer<Integer> deserializer = new IntegerDeserializer();
+
+            @Override
+            public void apply(byte[] key, byte[] value) {
+                keys.add(deserializer.deserialize("", key));
+            }
+        };
+    }
+
+    private static class MockRestoreConsumer extends MockConsumer<byte[], byte[]> {
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java]: Could we just edit MockConsumer directly? These overrides seem general enough to be incorporated into MockConsumer.

**Comment by guozhangwang** [comment]
Thanks for adding the ProcessorManagerTest! I was trying to do the same but your test coverage gets all I have here :)

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java (hunk 2 above)
Comment (ymatsuda) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java]: I don't think so. These overrides are rather specific to the usage of a restore consumer.

**Pull request merged**